source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,225 @@
+	}
+
+	const form = (
+		<form id={filterFormId}>
+			{createRadio(showFilterName, 'Show all', '', FilterSettings.ShowAll, true)}
+			{createRadio(showFilterName, 'Show only comments', 'Hides commits and events', FilterSettings.ShowOnlyComments, true)}
+			{createRadio(showFilterName, 'Show only unresolved comments', 'Also hides resolved reviews and hidden comments', FilterSettings.ShowOnlyUnresolvedComments, true)}
+			{showUnresolvedReviews}
+			<div className="Box-footer form-actions">
+				<button type="button" className="btn btn-primary" data-disable-with="Saving…" onClick={async () => saveSettings()}>Save</button>
+				<button type="reset" className="btn" data-close-dialog="">Cancel</button>
+			</div>
+		</form>
+	);
+
+	// This works on github enterprise - form is already preloaded
+	select('form', detailsDialog)?.remove();
+	// This works on normal github. Normally form is loaded in place of `include-fragment` after we open details dialog.
+	select('include-fragment', detailsDialog)?.remove();
+
+	detailsDialog.append(form);
+}
+
+function processTimelineItem(item: HTMLElement): void {
+	const pr = select('.js-comment[id^=pullrequestreview]', item);
+	const normalComment = select('.js-comment-container', item);
+
+	if (pr) {
+		processPR(item);
+	} else if (normalComment) {
+		applyDisplay(item, FilterSettings.ShowOnlyComments, FilterSettings.ShowOnlyUnresolvedComments);
+	} else {
+		applyDisplay(item, FilterSettings.ShowAll);
+	}
+}
+
+function processPR(item: HTMLElement): void {
+	let hasVisibleElement = false;
+
+	for (const threadContainer of select.all('.js-resolvable-timeline-thread-container', item)) {
+		const commentContainer = select('.inline-comment-form-container', threadContainer);
+
+		if (threadContainer.getAttribute('data-resolved') === 'true') {
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyComments);
+		} else if (commentContainer === null) {
+			// There is 1 special case here when github shows you a comment that was added to previous comment thread but it does not show whether it is resolved or not resolved comment.
+			// It's kinda tricky to know what to do with this so it is marked as normal comment for meantime.
+			// We are just checking here if user is able to comment inside that timeline thread, if not then it means we have this special situation that was just described.
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyComments, FilterSettings.ShowOnlyUnresolvedComments);
+		} else {
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyUnresolvedComments, FilterSettings.ShowOnlyUnresolvedReviews, FilterSettings.ShowOnlyComments);
+		}
+
+		// We need to hide whole thread group if we have hidden all comments inside.
+		hasVisibleElement = hasVisibleElement || threadContainer.style.display === '';
+	}
+
+	if (hasVisibleElement) {
+		item.style.display = '';
+	} else {
+		item.style.display = 'none';
+	}
```

Comment:
You can replace every usage of the `display` style with the `hidden` property, without if/else

```suggestion
	item.hidden = !hasVisibleElement
```

Reply:
I tried to do that but this was not working. It was not hiding/showing things.

Reply:
I just did, it works fine

Reply:
Hmm. Yeah it is working now. Weird it was not working for me.